api/src/files/files.controller.ts

```diff
@@ -0,0 +1,120 @@
+import { createReadStream, promises as fs, constants as fsConst } from 'node:fs';
+import { join } from 'node:path';
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  ParseIntPipe,
+  Patch,
+  Post,
+  Query,
+  Response,
+  StreamableFile,
+  UploadedFile,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Express, Response as Res } from 'express';
+import { JwtAuthGuard } from '../auth/jwt-auth.guard';
+import { uploadConfig } from '../config';
+import { PaginateDto } from '../posts/dto/paginate.dto';
+import { CurrentUser } from '../shared/decorators/current-user.decorator';
+import type { CustomPaginateResult } from '../shared/pagination';
+import { FileType } from '../shared/types/file-kinds.enum';
+import { User } from '../users/user.schema';
+import type { CreateFileUploadDto } from './dto/create-file-upload.dto';
+import type { CreateStudyDocDto } from './dto/create-study-doc.dto';
+import { UpdateStudyDocDto } from './dto/update-study-doc.dto';
+import type { StudyDoc } from './schemas/study-doc.schema';
+import { FilesService } from './services/file-uploads.service';
+import { StudyDocsService } from './services/study-docs.service';
+
+@UseGuards(JwtAuthGuard)
+@Controller({ path: 'files', version: '1' })
+export class FilesController {
+  constructor(private readonly studyDocsService: StudyDocsService, private readonly filesService: FilesService) {}
+
+  // Create
+  @UseInterceptors(FileInterceptor('file', { limits: { fileSize: uploadConfig.maxSize } }))
+  @Post('/study-docs/upload')
+  public async uploadCourseDocument(@CurrentUser() user: User,
+  @Body() body: Partial<CreateFileUploadDto & CreateStudyDocDto>,
+  @UploadedFile() file: Express.Multer.File): Promise<StudyDoc> {
```

Comment:
```suggestion
  public async uploadCourseDocument(
    @CurrentUser() user: User,
    @Body() body: Partial<CreateFileUploadDto & CreateStudyDocDto>,
    @UploadedFile() file: Express.Multer.File,
  ): Promise<StudyDoc> {
```
ESLint a décidé de faire la grève